### PR TITLE
feat: search by asset

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

- models use the new `search-by-asset-type` endpoint
- models assets have the `id` attribute so we should see the previews in the explorer and them into the project
- preparing code to transition in using the new search endpoint and to eventually deprecate `ResourceType`
- merged some data explorer cleanups/ui updates that I had in another branch

https://github.com/DARPA-ASKEM/terarium/assets/28237258/f2a7cae7-a919-4de9-bfc2-f6917b5d7545

### CODE REVIEW

File: packages/client/hmi-client/src/services/search.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import API from '@/api/api';
+import { AssetType } from '@/types/Types';
+
+export async function search(text: string, assetType: AssetType) {
+	const response = await API.get(`/search-by-asset-type/${assetType}?text=${text}`);
+	return response?.data ?? [];
+}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/services/search.ts:7]: Maybe a more specific name so things are less confusing down the road? "searchAsset"?

Does `text` need to be sanitized, url-encoded, escaped and whatnot? Not sure how we are processing it at the API level...

File: packages/client/hmi-client/src/services/search.ts
Comment (YohannParis) [inline packages/client/hmi-client/src/services/search.ts:1]: Please add the try catch to handle 204 and 4XX errors.

File: packages/client/hmi-client/src/services/search.ts (hunk 1 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/services/search.ts:7]: Maybe a `searchByAssetType`?

File: packages/client/hmi-client/src/components/navbar/tera-searchbar.vue [hunk 2]
```diff
@@ -28,13 +28,16 @@
...
 				<i class="pi pi-times clear-search" :class="{ hidden: !query }" @click="clearQuery" />
+				<!-- <Button
+					class="search-by-example-button"
+					icon="pi pi-upload"
+					text
+					rounded
+					size="small"
+					@click="searchByExampleToggle = !searchByExampleToggle"
+					:active="isSearchByExampleVisible"
+				/> -->
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/navbar/tera-searchbar.vue:39]:
```suggestion
```

File: packages/client/hmi-client/src/components/navbar/tera-searchbar.vue [hunk 3]
```diff
@@ -308,12 +313,12 @@ watch(
 }
 
 .p-input-icon-left {
-	margin-right: 1rem;
 	flex: 1;
 }
 
 i {
 	color: var(--text-color-subdued);
+	margin-left: 0.5rem;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/navbar/tera-searchbar.vue:321]:
```suggestion
	margin-left: var(--gap-small);
```

File: packages/client/hmi-client/src/page/data-explorer/components/tera-preview-panel.vue [hunk 4]
```diff
@@ -94,6 +89,7 @@ watch(
 		if (previewItem) {
 			previewItemState.value = previewItem;
 			previewItemResourceType.value = props.resourceType;
+			console.log(previewItem);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/data-explorer/components/tera-preview-panel.vue:92]:
```suggestion
```

File: packages/client/hmi-client/src/page/data-explorer/components/tera-search-results-list.vue [hunk 5]
```diff
@@ -306,15 +307,7 @@ ul {
...
-	color: var(--text-color-subdued);
-}
-
-.result-count {
-	font-size: var(--font-caption);
-}
-
-.result-count span {
-	color: var(--text-color-primary);
+	margin-left: 0.5rem;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/data-explorer/components/tera-search-results-list.vue:310]:
```suggestion
	margin-left: var(--gap-small);
```

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/client/hmi-client/src/services/search.ts [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
 import { AssetType } from '@/types/Types';
 
 export async function searchAsset(text: string, assetType: AssetType) {
-	const response = await API.get(`/search-by-asset-type/${assetType}?text=${text}`);
-	return response?.data ?? [];
+	try {
+		const response = await API.post(`/search-by-asset-type/${assetType}`, { text });
+		return response?.data ?? [];
+	} catch (error) {
+		logger.error(error);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/search.ts:13]: You should add a message. `Unable to fetch results from the question: ${text}`

**Pull request merged**